### PR TITLE
#70 Changes to monitor user group switching

### PR DESCRIPTION
**Pull request by Satendra-SR**
#70

### Types of changes

- [ ] Bug fix (a change which fixes an issue)
- [x] New feature (a change which adds functionality)

### Checklist:

- [x] I have performed a self-review of my own code.
- [x] The code follows the style guidelines of this project.
- [x] The code changes are passing the CI checks
- [x] I have documented my code wherever required.
- [x] The changes requires a change to the documentation.
- [ ] I have updated the documentation based on the my changes.

### CODE REVIEW

File: api/helpers/db_helper.py [hunk 1]
```diff
@@ -39,3 +39,11 @@ def get_program_prompt_id(jsonData):
 def save(data):
     db.session.add(data)
     db.session.commit()
+
+
+def get_user_by_phone(phone):
```
Comment (Satendra-SR) [inline api/helpers/db_helper.py:44]: We were using these functions quit often.

File: api/models/user_group.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from api.mixins import TimestampMixin
+from api import db, helpers
+from flask_sqlalchemy import BaseQuery
+
+
+class UserGroupQuery(BaseQuery):
+    def get_by_uuid(self, uuid):
+        return self.filter(UserGroup.group_uuid == uuid).first()
+
+    def get_unique(self, uuid, phone):
```
Comment (Satendra-SR) [inline api/models/user_group.py:10]: We are capturing group uuid from RP. So to get unique group details, we need two element. We are using user phone and group to get these details.

File: api/services/call_log_service.py [hunk 3]
```diff
@@ -113,17 +113,18 @@ def handle_parent_flow(self, jsonData):
...
-            parent_flow_data["parent_flow_name"] = parent_flow["name"]
-            missedcall_category_list = helpers.list_having_string(
-                parent_flow["name"], self.missedcall_flow_identifier
-            )
-            if missedcall_category_list:
-                """The call category is set to call back if missedcall flow has ran.
-                For that, the missed call flow name should contains string "missedcall"
-                """
-                self.call_category = models.CallLog.CallCategories.CALLBACK
+        if "parent" in jsonData and jsonData["parent"] is not None:
```
Comment (Satendra-SR) [inline api/services/call_log_service.py:116]: This was the issue with empty  `jsonData["parent"]` on production. When jsonData["parent"] was empty, was throwing `TypeError: argument of type 'NoneType' is not iterable`

File: api/services/prompt_service.py [hunk 4]
```diff
@@ -10,7 +10,11 @@ def __init__(self):
     def set_init_data(self, jsonData):
         user_phone = helpers.fetch_by_key("urn", jsonData["contact"])
         self.user_phone = helpers.sanitize_phone_string(user_phone)
-        flow_run_uuid = helpers.fetch_by_key("run_uuid", jsonData)
+        flow_run_uuid = helpers.fetch_by_key(
+            "run_uuid", jsonData
+        )  # Need to remove once we are done with making changes in webhooks
+        if "flow_run_details" in jsonData:
```
Comment (Satendra-SR) [inline api/services/prompt_service.py:16]: Fetching uuid from `flow_run_details` variable is available.

**Review by rudresh — APPROVED** [review]
Looks good.

**Pull request merged**